File: E1-two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        dicts = {}
        for index in range(len(nums)):
            num = nums[index]
            diff = target - num
            if diff in dicts:
                return [dicts[diff], index]
            else:
                dicts[num] = index

File: Java/E1-two-sum.java
class Solution {
  public int[] twoSum(int[] nums, int target) {
      HashMap<Integer, Integer> map = new HashMap<>();
      int[] res = new int[2];
      for(int i = 0; i < nums.length; i ++) {
          int diff = target - nums[i];
          if (map.containsKey(nums[i])) {
              res[0] = map.get(nums[i]);
              res[1] = i;
              break;
          } else {
              map.put(diff, i);
          }
      }
      return res;
  }
}